docs/source-2.0/java/client/codegen-integrations.rst

```diff
@@ -0,0 +1,80 @@
+====================
+Codegen Integrations
+====================
+
+Smithy Java provides a number of client codegen integrations that modify the code generated by the client codegen plugin.
+
+--------------
+Waiter Codegen
+--------------
+
+.. warning::
+
+    Only synchronous waiters are supported at this time.
+
+:ref:`Waiters <waiters>` are a client-side abstraction used to poll a resource until a desired state is reached,
+or until it is determined that the resource will never enter a desirable end state.
+Waiters can be defined in the Smithy model using the ``smithy.waiters#waitable`` trait.
+
+For example:
+
+.. code-block:: smithy
+   :caption: model.smithy
+
+    use smithy.waiters#waitable
+
+    @waitable(
+        BucketExists: {
+            documentation: "Wait until a bucket exists"
+            acceptors: [
+                {
+                    state: "success"
+                    matcher: {
+                        success: true
+                    }
+                }
+                {
+                    state: "retry"
+                    matcher: {
+                        errorType: "NotFound"
+                    }
+                }
+            ]
+        }
+    )
+    operation HeadBucket {
+        input: HeadBucketInput
+        output: HeadBucketOutput
+        errors: [NotFound]
+    }
+
+The waiter-codegen integration can be used to automatically generate waiters from waitable trait
+definitions in your Smithy model. To add the integration to your project (using the Smithy Gradle plugins):
```

Comment:
Using the waiters integration, you can automatically generate waiters from instances of the waitable trait in your Smithy model. If you are using the Smithy Gradle plugins, you can add this integration to your project like so: